Mutagen.Bethesda.Analyzers.Skyrim/Contextual/ConflictingVoiceTypesAnalyzer.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Mutagen.Bethesda.Analyzers.SDK.Analyzers;
+using Mutagen.Bethesda.Analyzers.SDK.Results;
+using Mutagen.Bethesda.Analyzers.SDK.Topics;
+using Mutagen.Bethesda.Plugins;
+using Mutagen.Bethesda.Skyrim;
+namespace Mutagen.Bethesda.Analyzers.Skyrim.Contextual;
+
+public class ConflictingVoiceTypesAnalyzer : IContextualAnalyzer
+{
+
+    public static readonly TopicDefinition<string?, int, string, string?> NpcsWithSameVoiceType = MutagenTopicBuilder.DevelopmentTopic(
+            "NPCs with the same voice type in same cell",
+            Severity.Suggestion)
+        .WithFormatting<string?, int, string, string?>("Cell {0} includes {1} npcs {2} with the same voice type {3}");
+
+    public IEnumerable<TopicDefinition> Topics { get; } = [NpcsWithSameVoiceType];
+
+    public ContextualAnalyzerResult Analyze(ContextualAnalyzerParams param)
+    {
+        var result = new ContextualAnalyzerResult();
+
+        foreach (var cell in param.LinkCache.PriorityOrder.WinningOverrides<ICellGetter>())
+        {
+            if (cell.IsExteriorCell()) continue;
+
+            var npcVoiceTypes = new Dictionary<INpcGetter, FormKey>();
```

Comment:
We'll still want to keep the editor id of the npcs at least so I'll downgrade the dictionary key to `IMajorRecordIdentifier`. Upgrading the value to `IFormLinkGetter<IVoiceTypeGetter>` seems good

Reply:
Not sure what you mean by `We'll still want to keep the editor id of the npcs`?

Are you saying you want to keep the EditorID as part of the key that determines collisions?  
I don't think swapping the key to `IMajorRecordIdentifier` will be the fix.  It won't reduce the things that are considered when determining key collisions, it will just restrict what can go into the key.  But whatever is put in there still uses its normal equality/hashcode... so it'll be essentially the same as it was before, but now you could put a weapon in there as well as a key.

I think either you'll want to do the `IFormLinkGetter<INpcGetter>` as a key, which requires `npc.ToFormLink()` when putting it in.  That's the fundamental difference, as the `ToFormLink` actually makes a FormLink class, and so the FormLink.Equals will be what's used for collisions.

But I'm remembering now that we have something even better:
`new Dictionary<INpcGetter, IFormLinkGetter<IVoiceTypeGetter>>(MajorRecord.FormKeyEqualityComparer)`
This lets you put the full object in as the key (for ease), but instructs the dictionary exactly how to consider collisions to just be the FormKey

Reply:
It was mostly needed to include the editor id to keep it around for the report but yeah you're totally right that this still has issues as a key. As I've added the possibility to pass form links into the topic which will be resolved when they are reported, I was able to fully remove this part from the analyzer.